gef.py

```diff
@@ -762,59 +987,64 @@ def __format__(self, format_spec):
         else:
             opcodes_len = int(format_spec[:-1])
 
-        opcodes_text = "".join("{:02x}".format(b) for b in self.opcodes[:opcodes_len])
+        opcodes_text = "".join(f"{b:02x}" for b in self.opcodes[:opcodes_len])
         if opcodes_len < len(self.opcodes):
             opcodes_text += "..."
-        return "{:#10x} {:{:d}} {:16} {:6} {:s}".format(self.address,
-                                                        opcodes_text,
-                                                        opcodes_len * 2 + 3,
-                                                        self.location,
-                                                        self.mnemonic,
-                                                        ", ".join(self.operands))
+        return (f"{self.address:#10x} {opcodes_text:{opcodes_len * 2 + 3:d}s} {self.location:16} "
+                f"{self.mnemonic:6} {', '.join(self.operands)}")
 
-    def __str__(self):
-        return "{:#10x} {:16} {:6} {:s}".format(
-            self.address, self.location, self.mnemonic, ", ".join(self.operands)
-        )
+    def __str__(self) -> str:
+        return f"{self.address:#10x} {self.location:16} {self.mnemonic:6} {', '.join(self.operands)}"
 
-    def is_valid(self):
+    def is_valid(self) -> bool:
         return "(bad)" not in self.mnemonic
 
 
 @lru_cache()
-def search_for_main_arena():
-    global __gef_current_arena__
-    malloc_hook_addr = parse_address("(void *)&__malloc_hook")
+def search_for_main_arena(to_string: bool = False) -> Union[int, str]:
+    """A helper function to find the libc `main_arena` address, either from symbol or from its offset
+    from `__malloc_hook`."""
+    try:
+        addr = int(parse_address(f"&{LIBC_HEAP_MAIN_ARENA_DEFAULT_NAME}"))
```

Comment:
```suggestion
        addr = parse_address(f"&{LIBC_HEAP_MAIN_ARENA_DEFAULT_NAME}")
```

`parse_address` already returns `int`

Reply:
It's for this kind of small things I'll be super happy to have type hinting! I'm sure there's plenty more